File: Leetcode/Python/Finished/Easy/ransom_note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        count = {}
        for char in ransomNote:
            if char in count:
                count[char] += 1
            else:
                count[char] = 1
        for char in magazine:
            if char in count:
                count[char] -= 1
        for key, value in count.items():
            if value > 0:
                return False
        return True

File: Leetcode/Java/src/Finished/Easy/ransom_note.java
import java.util.*;

class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        Map<Character, Integer> table = new HashMap<>();
        for (int i = 0; i < magazine.length(); i++) {
            char key = magazine.charAt(i);
            if (table.containsKey(key)) {
                table.put(key, table.get(key) + 1);
            } else {
                table.put(key, 1);
            }
        }
        for (int i = 0; i < ransomNote.length(); i++) {
            char key = ransomNote.charAt(i);
            if (table.containsKey(key)) {
                table.put(key, table.get(key) - 1);
                if (table.get(key) < 0) {
                    return false;
                }
            } else {
                return false;
            }
        }
        return (true);
    }
}